vhdl_lang/src/analysis/declarative.rs

```diff
@@ -695,6 +695,14 @@ impl<'a> AnalyzeContext<'a> {
                 }
             };
 
+            if Some(*entity_class) != get_entity_class(ent) {
+                diagnostics.push(Diagnostic::error(
+                    designator,
+                    format!("{} is not of class {}", ent.describe(), entity_class),
+                ));
+                return Ok(());
+            }
+
             // Attributes affect the underlying entity and cannot be set directly on aliases
             let ent = ent.as_actual();
```

Comment:
Blocking: Moving the entity class check above this line makes it no longer see through aliases.